src/taipy/core/_version/_version_fs_repository.py

```diff
@@ -10,49 +10,32 @@
 # specific language governing permissions and limitations under the License.
 
 import json
-from datetime import datetime
-from typing import List, Optional
+from typing import List
 
-from taipy.config import Config
 from taipy.logger._taipy_logger import _TaipyLogger
 
-from .._repository._filesystem_repository import _FileSystemRepository
 from ..exceptions.exceptions import VersionIsNotProductionVersion
-from ._version import _Version
 from ._version_model import _VersionModel
+from ._version_repository import _VersionRepository
 
 
-class _VersionFSRepository(_FileSystemRepository):
-    _LATEST_VERSION_KEY = "latest_version"
-    _DEVELOPMENT_VERSION_KEY = "development_version"
-    _PRODUCTION_VERSION_KEY = "production_version"
-
+class _VersionFSRepository(_VersionRepository):
     def __init__(self):
-        super().__init__(_VersionModel, "version", self._to_model, self._from_model)
+        super().__init__(model=_VersionModel, dir_name="version")
 
     @property
-    def _version_file_path(self):
-        return self.dir_path / "version.json"
-
-    def _to_model(self, version: _Version):
-        return _VersionModel(
-            id=version.id, config=Config._to_json(version.config), creation_date=version.creation_date.isoformat()
-        )
+    def dir_path(self):
+        return super().repository.dir_path
 
-    def _from_model(self, model):
-        version = _Version(id=model.id, config=Config._from_json(model.config))
-        version.creation_date = datetime.fromisoformat(model.creation_date)
-        return version
-
-    def _load_all(self, version_number: Optional[str] = "all") -> List[_Version]:
-        """Only load the json file that has "id" property.
+    @property
+    def _version_file_path(self):
+        return super().repository._storage_folder / "version.json"
```

Comment:
*Note*: On FS repo, the dev, production, and latest versions now stored at `.data/version.json`.